central/graphql/resolvers/node_components.go

```diff
@@ -97,11 +133,184 @@ func (resolver *Resolver) NodeComponentCount(ctx context.Context, args RawQuery)
 
 		return resolver.componentCountV2(ctx, RawQuery{Query: &query})
 	}
-	// TODO : Add postgres support
-	return 0, errors.New("Resolver NodeComponentCount does not support postgres yet")
+
+	if err := readNodes(ctx); err != nil {
+		return 0, err
+	}
+	query, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return 0, err
+	}
+	loader, err := loaders.GetNodeComponentLoader(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	return loader.CountFromQuery(ctx, query)
 }
 
+/*
+Utility Functions
+*/
+
 func queryWithNodeIDRegexFilter(q string) string {
 	return search.AddRawQueriesAsConjunction(q,
 		search.NewQueryBuilder().AddRegexes(search.NodeID, ".+").Query())
 }
+
+func (resolver *nodeComponentResolver) withNodeComponentScope(ctx context.Context) context.Context {
+	if features.PostgresDatastore.Enabled() {
+		return scoped.Context(ctx, scoped.Scope{
+			Level: v1.SearchCategory_NODE_COMPONENTS,
+			ID:    resolver.data.GetId(),
+		})
+	}
+	return scoped.Context(ctx, scoped.Scope{
+		Level: v1.SearchCategory_IMAGE_COMPONENTS,
+		ID:    resolver.data.GetId(),
+	})
+}
+
+func (resolver *nodeComponentResolver) nodeComponentQuery() *v1.Query {
+	return search.NewQueryBuilder().AddExactMatches(search.ComponentID, resolver.data.GetId()).ProtoQuery()
+}
+
+func (resolver *nodeComponentResolver) nodeComponentRawQuery() string {
+	return search.NewQueryBuilder().AddExactMatches(search.ComponentID, resolver.data.GetId()).Query()
+}
+
+/*
+Sub Resolver Functions
+*/
+
+// FixedIn returns the node component version that fixes all the fixable vulnerabilities in this component.
+func (resolver *nodeComponentResolver) FixedIn(ctx context.Context) string {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.NodeComponents, "FixedIn")
+	return ""
+}
+
+// NodeComponentLastScanned is the last time the node component was scanned in a node.
+func (resolver *nodeComponentResolver) NodeComponentLastScanned(ctx context.Context) (*graphql.Time, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.NodeComponents, "NodeComponentLastScanned")
+	nodeLoader, err := loaders.GetNodeLoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	componentQuery := resolver.nodeComponentQuery()
+	componentQuery.Pagination = &v1.QueryPagination{
+		Limit:  1,
+		Offset: 0,
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field:    search.NodeScanTime.String(),
+				Reversed: true,
+			},
+		},
+	}
+
+	nodes, err := nodeLoader.FromQuery(ctx, componentQuery)
+	if err != nil || len(nodes) == 0 {
+		return nil, err
+	} else if len(nodes) > 1 {
+		return nil, errors.New("multiple nodes matched for last scanned component query")
+	}
+
+	return timestamp(nodes[0].GetScan().GetScanTime())
+}
+
+// License of the node component
+func (resolver *nodeComponentResolver) License(_ context.Context) (*licenseResolver, error) {
```

Comment:
This is not relevant to nodes. We should work closely with Van to remove this fields from FE. For now, keep it to avoid breaking changes but make sure to remove in 72. Add a task to the follow-up epic https://issues.redhat.com/browse/ROX-10992